File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/di/component/ApplicationComponent.java
package tisoul.dev.androidarchitecture.di.component;


import android.content.Context;

import javax.inject.Singleton;

import dagger.Component;
import tisoul.dev.androidarchitecture.di.module.ApplicationModule;
import tisoul.dev.domain.repository.ProfileRepository;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {
    Context context();

    ProfileRepository profileRepository();
}

File: AndroidArchitecture/app/src/main/java/tisoul/dev/androidarchitecture/di/component/ApplicationComponent.kt
package tisoul.dev.androidarchitecture.di.component


import android.content.Context
import dagger.Component
import tisoul.dev.androidarchitecture.di.module.ApplicationModule
import tisoul.dev.domain.repository.ProfileRepository
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {
    fun context(): Context

    fun profileRepository(): ProfileRepository
}
